spring-kafka/src/test/java/org/springframework/kafka/config/KafkaListenerEndpointRegistryTests.java

```diff
@@ -47,4 +63,77 @@ void unregister() {
 		assertThat(unregistered).isSameAs(container);
 	}
 
+	@DisplayName("getListenerContainersMatching throws on null predicate")
+    @Test
+    void getListenerContainersMatchingThrowsOnNullPredicate() {
+		// Given
+        KafkaListenerEndpointRegistry registry = new KafkaListenerEndpointRegistry();
+		// When
+		// Then
+        assertThatIllegalArgumentException()
+			.isThrownBy(() -> registry.getListenerContainersMatching(null))
+            .withMessage("'idMatcher' cannot be null");
+    }
+
+	@DisplayName("getListenerContainersMatching should return unmodifiable list")
+	@Test
+    void testGetListenerContainersMatchingReturnsUnmodifiableList() {
+        // Given
+        KafkaListenerEndpointRegistry registry = new KafkaListenerEndpointRegistry();
+        registerListenerWithId(registry, "foo");
+        // When
+        Collection<MessageListenerContainer> listeners = registry.getListenerContainersMatching(s -> true);
+        // Then
+        assertThrows(UnsupportedOperationException.class, () -> listeners.add(mock(MessageListenerContainer.class)));
```

Comment:
Why don't use `assertThatExceptionOfType(UnsupportedOperationException.class)` instead?

It also probably going to fail on Checkstyle since we have a limited set of allowed `static` method imports.

Reply:
Right, looking now there is no pre-existing usage of `assertThatExceptionOfType()`